src/driver/sqljs/SqljsDriver.ts

```diff
@@ -0,0 +1,198 @@
+                this.createDatabaseConnectionWithImport(JSON.parse(localStorageContent));
+            }
+        }
+        else {
+            this.createDatabaseConnectionWithImport(fileNameOrLocalStorageOrData);
+        }
+    }
+
+    /**
+     * saved the current database to the given file (node.js) or local storage key (browser)
+     */
+    async save(location?: string) {
+        if (!location && !this.options.location) {
+            throw new Error(`No location is set, specify a location parameter or add the location option to your configuration`);
+        }
+        
+        let path = "";
+        if (location) {
+            path = location;
+        }
+        else if (this.options.location) {
+            path = this.options.location;
+        }
+
+        if (PlatformTools.type === "node") {
+            try {
+                await PlatformTools.writeFile(path, this.databaseConnection.export());
+            }
+            catch (e) {
+                throw new Error(`Could not save database, error: ${e}`);
+            }
+        }
+        else {
+            const content = JSON.stringify(this.databaseConnection.export());
+            PlatformTools.getGlobalVariable().localStorage.setItem(path, content);
+        }
+    }
+    
+    /**
+     * returns the current database as Uint8Array
+     */
+    export(): Uint8Array {
+        return this.databaseConnection.export();
+    }
+
+    // -------------------------------------------------------------------------
+    // Protected Methods
+    // -------------------------------------------------------------------------
+
+    /**
+     * Creates connection with the database.
+     */
+    protected createDatabaseConnection() {
+        if (this.options.location) {
+            this.load(this.options.location);
+        }
+        else {
+            this.createDatabaseConnectionWithImport(this.options.database);
+        }
+    }
+
+    protected createDatabaseConnectionWithImport(database?: Uint8Array) {
+        return new Promise<void>((ok, fail) => {
+            if (database && database.length > 0) {
+                this.databaseConnection = new this.sqlite.Database(database);
+            }
+            else {
+                this.databaseConnection = new this.sqlite.Database();
+            }
+            this.startTimer();
+            ok(this.databaseConnection);
+        });
+    }
+
+    /**
+     * If driver dependency is not given explicitly, then try to load it via "require".
+     */
+    protected loadDependencies(): void {
+        if (PlatformTools.type === "browser") {
+            this.sqlite = window.SQL;
+        }
+        else {
+            try {
+                this.sqlite = PlatformTools.load("sql.js");
+
+            } catch (e) {
+                throw new DriverPackageNotInstalledError("sql.js", "sql.js");
+            }
+        }
+    }
+
```

Comment:
make those methods protected too and put comments

Reply:
I will add comments, but they are public because I have to call them from the entity manager